irene-perez/precurso/tema2/airlines_pro_Irene.js

```diff
@@ -0,0 +1,210 @@
+        case ">":
+            return array.filter(function (element) {
+                return element[property] > evaluationValue;
+            });
+        case "=":
+            return array.filter(function (element) {
+                return element[property] === evaluationValue;
+            });
+    }
+}
+//Función que define el formato del texto a imprimir en consola para que sea amigable para el usuario
+function textFormatter(flight) {
+    return `El vuelo con origen: ${flight.from}, y destino: ${flight.to}, tiene un precio total de ${flight.cost}€. Además, ${(flight.scale) ? " sí realiza escalas" : "no realiza ninguna escala"}.`;
+}
+//Función del usuario
+function userFunction() {
+    let searchPrice = prompt('Introduzca el precio del vuelo que desea buscar:');
+    //Para que las equivalencias tengan sentido, para que la string no valga más que un número y se comparen números con números
+    searchPrice = parseFloat(searchPrice);
+    //Si lo que introduce no es un número
+    if (!searchPrice) {
+        console.log('Le has dado a cancelar, ¡hasta otra!');
+    } else if (isNaN(searchPrice)) {
+        console.log('No has introducido ningún número.');
+        return userFunction();
+    } else if (typeof searchPrice === 'number') {
+        let samePrice = priceEvaluation(flights, searchPrice, "=", "cost").map(function (flight) { return textFormatter(flight) });
+        //Usamos el método ternario por si el precio introducido no coincide con ningún otro del array flights, por eso se pone una condición y si esta se cumple o no
+        samePrice.length ?
+            console.log(`Los vuelos con los precios igual a ${searchPrice} son los siguientes:\n${samePrice.join('\n')}`)
+            //Se usa el método JSON.stringify para convertir el array en una string y así imrpimirla como tal
+            :
+            console.log('No hay vuelos con precio igual al introducido.');
+
+        //Se imprimen los vuelos con los precios más altos que el introducido
+        let higherPrice = priceEvaluation(flights, searchPrice, ">", "cost").map(function (flight) { return textFormatter(flight) });
+        higherPrice.length ?
+            console.log(`Los vuelos con los precios más altos que ${searchPrice} son:\n${higherPrice.join('\n')}`)
+            :
+            console.log('No hay vuelos con precio más alto al introducido.')
+
+        //Se imprimen los vuelos con los precios más bajos que el introducido
+        let lowerPrice = priceEvaluation(flights, searchPrice, "<", "cost").map(function (flight) { return textFormatter(flight) });
+        lowerPrice.length ?
+            console.log(`Los vuelos con los precios más bajos que ${searchPrice} son:\n${lowerPrice.join('\n')}`)
+            :
+            console.log('No hay vuelos con precio más bajo al introducido.');
+
+        //La id del vuelo que el usuario quiere comprar
+        let buyFlight = parseInt(prompt('Introduzca el id del vuelo que quiere comprar por favor.'));
```

Comment:
It is better (and a good practice) to first declare the variables to later use them.

Declare at the beginning of the function all 'lets', the use it and assign values to them.